pandas/core/groupby/ops.py

```diff
@@ -494,7 +495,7 @@ def _cython_operation(
             else:
                 values = ensure_int_or_float(values)
         elif is_numeric and not is_complex_dtype(values):
-            values = ensure_float64(values)
+            values = ensure_float64(ensure_float(values))
```

Comment:
hmm, we really need to do this twice?

Reply:
I did it this way because we need to have `ensure_float` to convert a EA FloatingArray into a numpy float array, and the cython `ensure_float64` is written to expect a numpy array, not an EA. 
If you actually have already a float64 ndarray, the `ensure_float` is a no-op.

Reply:
how about you make the same mod in ensure_float64 then, much more clear

Reply:
`ensure_float64` is a cython function that is used in many places, so doing something like that might affect performance in many cases, so needs to be done carefully, so I would rather want to keep that for a different PR. I also think it might be clearn to leave those `ensure_<dtype>` to deal with only numpy arrays (in addition, `ensure_float64` is also implemented as a template-ized method for many different dtypes, so it's not that straightforward to add a custom check only for float).